### DISCUSSION TITLE
Plugin instance --no-fail option: job failed but I want to keep my data

### DISCUSSION THREAD

**Discussion by jennydaman** [FNNDSC/CHRIS_docs#21] [Ideas]
@rudolphpienaar mentioned: what if a plugin instance runs for 5 hours, fails, but I want to keep my data?

tl;dr plugin developers should add a `--no-fail` argument.

An interpretation and solution to this within the (current) confines of the system would be: _ChRIS_ itself is going to be strict: `finishedSuccessfully` can be interpreted to mean that the output files should be kept, hence `finishedWithError` means the output files are not useful anyways since it's the result of a failure, By this interpretation, it can be said that it's the plugin developer's decision to make. A plugin developer may add the option `--no-fail` (or something alike) to their plugin, which means "keep output even if task is unsuccessful." Next, the user must run the plugin instance giving `--no-fail` explicitly, implying "I realize the consequence that the plugin instance's output data is now a union type where it might be the result of a successful task OR the result of an unsuccessful task and I'm going to handle this union type appropriately."

Of course the plugin instance might still fail despite the `--no-fail` flag, which means the task failed in an unexpected way, in which case it's implied that there's nothing the user can do about it — they must report the bug to the plugin developer who should fix their plugin.

Example:

https://github.com/FNNDSC/ep-interpolate-surface-with-sphere/blob/60f65ddb55444603b4fc3e79b0b277e57b3f0f02/isws.py#L27-L28

**Comment by rudolphpienaar** [comment]
This is a really good suggestion! I'm wondering about the flag a bit. Perhaps the behaviour should shift to _always_ pulling output and instead of a `--no-fail` it might be useful to have a `--output-only-no-fail`. Ok, that's cumbersome... just thinking out loud.
